meshroom/nodes/aliceVision/SketchfabUpload.py

```diff
@@ -0,0 +1,200 @@
+        self._callback = callback
+        self._cb_args = cb_args
+        self._cb_kwargs = cb_kwargs
+        self._progress = 0
+        self._len = len(buf)
+        io.BytesIO.__init__(self, buf)
+
+    def __len__(self):
+        return self._len
+
+    def read(self, n=-1):
+        chunk = io.BytesIO.read(self, n)
+        self._progress += int(len(chunk))
+        self._cb_kwargs.update({
+            'size'    : self._len,
+            'progress': self._progress
+        })
+        if self._callback:
+            try:
+                self._callback(*self._cb_args, **self._cb_kwargs)
+            except Exception as e: # catches exception from the callback
+                self._cb_kwargs['logManager'].logger.warning('Error at callback: {}'.format(e))
+        return chunk
+
+def progressUpdate(size=None, progress=None, logManager=None):
+    if not logManager.progressBar:
+        logManager.makeProgressBar(size, 'Upload progress:')
+
+    logManager.updateProgressBar(progress)
+
+class SketchfabUpload(desc.Node):
+    size = desc.DynamicNodeSize('inputFiles')
+    inputs = [
+        desc.ListAttribute(
+            elementDesc=desc.File(
+                name="input",
+                label="Input",
+                description="",
+                value="",
+                uid=[0],
+            ),
+            name="inputFiles",
+            label="Input Files",
+            description="Input Files to export.",
+            group="",
+        ),
+        desc.ChoiceParam(
+            name='maxSize',
+            label='Maximum Upload Size',
+            description='The maximum upload size in MB.',
+            value=50,
+            values=(50, 200, 500),
+            exclusive=True,
+            uid=[0],
+            advanced=True,
+        ),
+        desc.BoolParam(
+            name='limitSize',
+            label='Limit Upload Size',
+            description='Change maximum download size in advanced attributes.',
+            value=True,
+            uid=[0],
+        ),
+        desc.StringParam(
+            name='apiToken',
+            label='API Token',
+            description='Get your token from https://sketchfab.com/settings/password',
+            value='',
+            uid=[0],
+        ),
+        desc.StringParam(
+            name='title',
+            label='Title',
+            description='Title cannot be longer than 48 characters.',
+            value='',
+            uid=[0],
+        ),
+        desc.StringParam(
+            name='description',
+            label='Description',
+            description='Description cannot be longer than 1024 characters.',
+            value='',
+            uid=[0],
+        ),
+        desc.StringParam(
+            name='license',
+            label='License',
+            description='License label.',
+            value='CC Attribution',
+            uid=[0],
+        ),
+        desc.BoolParam(
+            name='isPublished',
+            label='Publish',
```

Comment:
I think it would be better to save the model in [Draft Mode](https://help.sketchfab.com/hc/en-us/articles/115000472906-Draft-Mode) by default. You likely want to fix the model orientation and lighting before publishing.

Reply:
The default value is already False.

Reply:
I see, missed this.